app/models/legacy/dataset.rb

```diff
@@ -0,0 +1,67 @@
+class Legacy::Dataset < SimpleDelegator
+
+  def metadata_json
+    organisation = Organisation.find(organisation_id)
+    ckan_dataset = {
+      "id" => uuid,
+      "name" => name,
+      "title" => title,
+      "notes" => summary,
+      "description" => summary,
+      "organization" => { "name" => organisation.name },
+      "update_frequency" => convert_freq_to_legacy_format(frequency),
+      "update_frequency-other" => custom_frequency,
+      "extras" => [{"key" => "update_frequency",
+                    "package_id" => uuid,
+                    "value" => convert_freq_to_legacy_format(frequency)}
+                  ],
+      "unpublished" => unpublished?(status),
+      "metadata_created" => created_at,
+      "metadata_modified" => last_updated_at,
+      "geographic_coverage" => [(location1 || "").downcase],
+      "license_id" => licence
+    }.compact
+    extend_extras(ckan_dataset).to_json
+  end
+
+  def update
+    Legacy::Server.new.update(metadata_json)
+  end
+
+  private
+
+  def extend_extras(ckan_dataset)
+    if ckan_dataset["update_frequency"] == 'other'
+      extra = {
+        "key" => "update_frequency-other",
+        "package_id" => uuid,
+        "value" => ckan_dataset["update_frequency-other"]
+      }
+      ckan_dataset["extras"].push(extra)
+    end
+    ckan_dataset
+  end
+
+  def unpublished?(string)
+    string == "draft"
```

Comment:
@hannako the day we add another status (like "discarded"), this will return `true`. The only way to tell whether a dataset is unpublished is if its status is anything else than `"published"`.